src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -5786,38 +5786,53 @@ public void disableAuditLog()
         logger.info("Auditlog is disabled");
     }
 
+    @Deprecated
     public void enableAuditLog(String loggerName, String includedKeyspaces, String excludedKeyspaces, String includedCategories, String excludedCategories,
                                String includedUsers, String excludedUsers) throws ConfigurationException, IllegalStateException
     {
-        enableAuditLog(loggerName, Collections.emptyMap(), includedKeyspaces, excludedKeyspaces, includedCategories, excludedCategories, includedUsers, excludedUsers);
+        enableAuditLog(loggerName, Collections.emptyMap(), includedKeyspaces, excludedKeyspaces, includedCategories, excludedCategories, includedUsers, excludedUsers,
+                       Integer.MIN_VALUE, null, null, Long.MIN_VALUE, Integer.MIN_VALUE, null);
+    }
+
+    public void enableAuditLog(String loggerName, String includedKeyspaces, String excludedKeyspaces, String includedCategories, String excludedCategories,
+                               String includedUsers, String excludedUsers, Integer maxArchiveRetries, Boolean block, String rollCycle,
+                               Long maxLogSize, Integer maxQueueWeight, String archiveCommand) throws ConfigurationException, IllegalStateException
+    {
+        enableAuditLog(loggerName, Collections.emptyMap(), includedKeyspaces, excludedKeyspaces, includedCategories, excludedCategories, includedUsers, excludedUsers,
+                       maxArchiveRetries, block, rollCycle, maxLogSize, maxQueueWeight, archiveCommand);
     }
 
+    @Deprecated
     public void enableAuditLog(String loggerName, Map<String, String> parameters, String includedKeyspaces, String excludedKeyspaces, String includedCategories, String excludedCategories,
```

Comment:
Hey, I just noticed the upper method, not this one is a non-used one and I guess that one had to be actually deprecated?

Reply:
no no, that one is supposed to be used.

Reply:
There is one unused enableAuditLog method that is not deprecated and I am not sure why, can you, please, double check?

Reply:
that one does not accept Map, for tooling like JConsole, there is not any way how to invoke a method if it accepts Map, it is not invokable there.

Reply:
People tend to use tooling different from nodetool and they are not call that programmatically either, if there is just one method accepting paramter like Map, we just make it incallable in that tooling. JConsole, for example, are used quite heavily with our devops and they are more often in JConsole than nodetool (some of them) so I just do not cut the off completely here.

I know we might now try to figure out some workaround or similar but that would end up being even more awkward and custom and that is not the problem I want to address in this ticket.

Methods in interfaces are very inconsistent and they lack a robust and coherent approach and us spending any more time on this is rather quesionable, when it comes to this ticket itself and what we try to achieve.

Reply:
This specific method was deprecated because it does not contain all other parameters we enabled to be specified. The longer version is used instead.

Reply:
Ok, so just add a comment on commit so that people don't have to ask, please. Thanks